common/warp_utils/src/json.rs

```diff
@@ -0,0 +1,22 @@
+use bytes::{Buf, Bytes};
+use serde::de::DeserializeOwned;
+use std::error::Error as StdError;
+use warp::{Filter, Rejection};
+
+use crate::reject;
+
+struct Json;
+
+type BoxError = Box<dyn StdError + Send + Sync>;
+
+impl Json {
+    fn decode<B: Buf, T: DeserializeOwned>(mut buf: B) -> Result<T, BoxError> {
+        serde_json::from_slice(&buf.copy_to_bytes(buf.remaining())).map_err(Into::into)
```

Comment:
I think `Bytes` dereferences to `&[u8]`

```suggestion
        serde_json::from_slice(&buf).map_err(Into::into)
```

Reply:
This is a `Buf` which I dont think deferences directly to `&[u8]`. I think the other option here is `&buf.chunk()`

Reply:
ah I think I can just pass in bytes directly, I see what you're saying now